# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/ron/tests/449_tagged_enum.rs
# Commit: 3f167f8b3137
# Full Hash: 3f167f8b313749f50f135e189e5854bef7ac0c73
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-10-03 18:08 +0000
# Regressor Bug: 1991551
# File Overlap Count: 23
# Overlapping Files: third_party/rust/wgpu-hal/Cargo.toml, supply-chain/audits.toml, third_party/rust/naga/src/back/spv/writer.rs, third_party/rust/naga/src/back/msl/writer.rs, third_party/rust/wgpu-types/src/lib.rs
# Description:
#   Bug 1991551 - build(webgpu): update WGPU to 8c4aebc0c1b75e3f7412dfc180745a0717fc30ec r=webgpu-reviewers,supply-chain-reviewers,nical
# ==============================================================================

diff -r 823439f307a0 -r 3f167f8b3137 third_party/rust/ron/tests/449_tagged_enum.rs
--- a/third_party/rust/ron/tests/449_tagged_enum.rs	Fri Oct 03 18:01:30 2025 +0000
+++ b/third_party/rust/ron/tests/449_tagged_enum.rs	Fri Oct 03 18:08:29 2025 +0000
@@ -43,16 +43,29 @@
 
 #[test]
 fn test_serde_content_hack() {
-    assert_eq!(
+    assert!(matches!(
         std::any::type_name::<serde::__private::de::Content>(),
-        "serde::__private::de::content::Content"
-    );
+        "serde::__private::de::content::Content" | "serde::__private::de::content::Content<'_>"
+    ));
 }
 
 #[test]
 fn test_serde_internally_tagged_hack() {
-    const SERDE_CONTENT_CANARY: &str = "serde::__private::de::content::Content";
-    const SERDE_TAG_KEY_CANARY: &str = "serde::__private::de::content::TagOrContent";
+    // ensure that these are the same as in ron::de module
+    fn is_serde_content<T>() -> bool {
+        matches!(
+            core::any::type_name::<T>(),
+            "serde::__private::de::content::Content" | "serde::__private::de::content::Content<'_>"
+        )
+    }
+
+    fn is_serde_tag_or_content<T>() -> bool {
+        matches!(
+            core::any::type_name::<T>(),
+            "serde::__private::de::content::TagOrContent"
+                | "serde::__private::de::content::TagOrContent<'_>"
+        )
+    }
 
     struct Deserializer {
         tag_key: Option<String>,
@@ -87,7 +100,7 @@
         where
             K: serde::de::DeserializeSeed<'de>,
         {
-            assert_eq!(std::any::type_name::<K::Value>(), SERDE_TAG_KEY_CANARY);
+            assert!(is_serde_tag_or_content::<K::Value>());
 
             if let Some(tag_key) = self.tag_key.take() {
                 return seed
@@ -109,11 +122,11 @@
             V: serde::de::DeserializeSeed<'de>,
         {
             if self.field_key.is_some() {
-                assert_ne!(std::any::type_name::<V::Value>(), SERDE_CONTENT_CANARY);
+                assert!(!is_serde_content::<V::Value>());
                 return seed.deserialize(serde::de::value::StrDeserializer::new(&self.tag_value));
             }
 
-            assert_eq!(std::any::type_name::<V::Value>(), SERDE_CONTENT_CANARY);
+            assert!(is_serde_content::<V::Value>());
 
             seed.deserialize(serde::de::value::I32Deserializer::new(self.field_value))
         }